src/attackmate/executors/baseexecutor.py

```diff
@@ -97,7 +97,10 @@ def log_json(self, logger: logging.Logger, command, time):
             if key not in command_dict:
                 command_dict['parameters'][key] = value
 
-        logger.info(json.dumps(command_dict))
+        try:
+            logger.info(json.dumps(command_dict))
+        except TypeError:
+            logger.warn('log_json is broken')
```

Comment:
looks like for the new LoopCommand `for key, value command.__dict__.items() `processes the item command.commands, which is a list with items of type Command (and json.dump does not work on Objects, only basic data types). you could exclude command.commands from the logging, and log every command in the loop on its own anyway? or convert into serializable format

Reply:
I'll leave it open. lets fix that in a separate pr